staging/src/k8s.io/code-generator/generate-groups.sh

```diff
@@ -47,10 +47,7 @@ GROUPS_WITH_VERSIONS="$4"
 shift 4
 
 (
-  # To support running this script from anywhere, we have to first cd into this directory
-  # so we can install the tools.
-  cd "$(dirname "${0}")"
-  go install ./cmd/{defaulter-gen,client-gen,lister-gen,informer-gen,deepcopy-gen}
+  go install k8s.io/code-generator/cmd/{defaulter-gen,client-gen,lister-gen,informer-gen,deepcopy-gen}
```

Comment:
this change might break our CI.
will defer to the code-gen maintainers though.

/ok-to-test

Reply:
> this change might break our CI.
> will defer to the code-gen maintainers though.
> 
> /ok-to-test

Tried my best to resolve this by gating this on users explicitly opting into `GO111MODULES=on`. If you have any other ideas on how we can handle this, let me know!

Reply:
locally forcing GOMODULE + vendor mode is probably a better option, see https://github.com/kubernetes/kubernetes/pull/85657/files

Reply:
I enabled `GO111MODULES=on` in `hack/update-codegen.sh` and `hack/verify-codegen.sh` to enable it for this project. We don't want to force this in generate-groups.sh, because then you would be forcing modules on any project that wants to depend on this script (incl. non-module based projects).

For example, [Tekton currently uses this script with dep](https://github.com/tektoncd/pipeline/blob/master/hack/update-codegen.sh#L31) (we're in the process of migrating to modules). Forcing modules on would likely cause issues, since there is no go.mod in the parent project.

I think it makes more sense for project's to opt in for whether to use the module based approach or not, and should also decide whether they want to include the vendor directory or not.